lib/ios-log.js

```diff
@@ -0,0 +1,203 @@
+
+let START_TIMEOUT = 10000;
+let logger = getLogger('IOS-log');
+
+class IOSLog { 
+
+  constructor(opts) {
+    this.udid = opts.udid;
+    this.simUdid = opts.simUdid;
+    this.showLogs = opts.showLogs;
+    this.proc = null;
+    this.onIosLogStart = null;
+    this.iosLogStarted = false;
+    this.iosLogStartTime = null;
+    this.loggingModeOn = true;
+    this.logs = [];
+    this.logRow = '';
+    this.logsSinceLastRequest = [];
+    this.startTimer = null;
+  }
+
+  async startCapture () {
+    if (this.udid) { 
+      this.loggingModeOn = false;
+      let spawnEnv = _.clone(process.env);
+      logger.debug('Attempting iOS device log capture via libimobiledevice idevicesyslog');
+      let err = await which('idevicesyslog');
+      if (!err) { 
+        this.proc = new SubProcess('idevicesyslog', {env: spawnEnv});
+      } else {
+        logger.warn('Could not capture device log using libimobiledevice idevicesyslog.' +
+            ' Libimobiledevice probably is not installed');
+        logger.debug('Attempting iOS device log capture via deviceconsole');
+        let limDir = path.resolve(__dirname, '../../../build/deviceconsole');
+        spawnEnv.PATH = `${process.env.PATH}:${limDir}`;
+        spawnEnv.DYLD_LIBRARY_PATH = `${limDir}:${process.env.DYLD_LIBRARY_PATH}`;
+        this.proc =  new SubProcess('deviceconsole', ['-u', this.udid], {env: spawnEnv});
+      }
+      await this.finishStartingLogCapture();
+      return;
+    } 
+
+    let xcodeVersion = await xcode.getVersion();
+    let ver = parseInt(xcodeVersion.split('.')[0], 10);
+
+    if ( ver < 5 ) {
+      logger.debug('Starting iOS6.* simulator log capture');
+      this.proc = new SubProcess('tail', ['-f', '-n', '1', '/var/log/system.log']);
+      await this.finishStartingLogCapture();
+      return;
+    }
+
+    let logsPath;
+
+    if (ver >= 6) { 
+      logger.debug('Starting iOS 8.*/9.* simulator log capture');
+        if (_.isUndefined(this.simUdid)) {
+          logger.errorAndThrow('iOS 8*/9.* log capture requires a sim udid'); 
+        }
+        logsPath = path.resolve(process.env.HOME, 'Library', 'Logs', 'CoreSimulator', this.simUdid); 
+    } else {
+        logger.debug('Starting iOS7.* simulator log capture');
+        logsPath = path.resolve(process.env.HOME, 'Library', 'Logs', 'iOS Simulator', '7.*');
+    }
+
+    try {
+      if (logsPath.indexOf('*') >= 0) {
+        return; 
+      }
+      await mkdirp(logsPath);
+      await touch(path.resolve(logsPath, 'system.log'));
+      await fs.appendFile(path.resolve(logsPath, 'system.log'), 
+            'A new Simulator is about to start!'); 
+      let files;
+      try {
+        files = await glob(path.resolve(logsPath, 'system.log'));
+        if (files.length < 1) {
+          throw 'Could not start log capture';
+        }
+      } catch (e) {
+        logger.error(`Could not start log capture because no iOS ` +
+        `simulator logs could be found at ${logsPath}/system.log` +
```

Comment:
```js
logger.error(`Could not start log capture because no iOS ` +
             `simulator logs could be found at ${logsPath}/system.log` +
             `Logging will not be functional for this run`);
```